service/history/executor_factory.go

```diff
@@ -0,0 +1,295 @@
+	return NewArchivalQueueTaskExecutor(
+		f.Archiver,
+		shardCtx,
+		workflowCache,
+		f.RelocatableAttributesFetcher,
+		f.MetricsHandler,
+		logger,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTimerActiveExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	workflowDeleteManager deletemanager.DeleteManager,
+	logger log.Logger,
+) queues.Executor {
+	return newTimerQueueActiveTaskExecutor(
+		shardCtx,
+		workflowCache,
+		workflowDeleteManager,
+		logger,
+		f.MetricsHandler,
+		f.Config,
+		f.MatchingClient,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTimerStandbyExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	workflowDeleteManager deletemanager.DeleteManager,
+	nDCHistoryResender xdc.NDCHistoryResender,
+	logger log.Logger,
+	clusterName string,
+) queues.Executor {
+	return newTimerQueueStandbyTaskExecutor(
+		shardCtx,
+		workflowCache,
+		workflowDeleteManager,
+		nDCHistoryResender,
+		f.MatchingClient,
+		logger,
+		f.MetricsHandler,
+		clusterName,
+		f.Config,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTransferActiveExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	logger log.Logger,
+) queues.Executor {
+	return newTransferQueueActiveTaskExecutor(
+		shardCtx,
+		workflowCache,
+		f.ArchivalClient,
+		f.SdkClientFactory,
+		logger,
+		f.MetricsHandler,
+		f.Config,
+		f.MatchingClient,
+		f.VisibilityManager,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTransferStandbyExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	nDCHistoryResender xdc.NDCHistoryResender,
+	logger log.Logger,
+	clusterName string,
+) queues.Executor {
+	return newTransferQueueStandbyTaskExecutor(
+		shardCtx,
+		workflowCache,
+		f.ArchivalClient,
+		nDCHistoryResender,
+		logger,
+		f.MetricsHandler,
+		clusterName,
+		f.MatchingClient,
+		f.VisibilityManager,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateExecutorWrapper(
+	currentClusterName string,
+	activeExecutor queues.Executor,
+	standbyExecutor queues.Executor,
+	logger log.Logger,
+) queues.Executor {
+	return queues.NewExecutorWrapper(
+		currentClusterName,
+		f.NamespaceRegistry,
+		activeExecutor,
+		standbyExecutor,
+		logger,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateVisibilityExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	logger log.Logger,
+) queues.Executor {
+	return newVisibilityQueueTaskExecutor(
+		shardCtx,
+		workflowCache,
+		f.VisibilityManager,
+		logger,
+		f.MetricsHandler,
+		f.Config.VisibilityProcessorEnsureCloseBeforeDelete,
+		f.Config.VisibilityProcessorEnableCloseWorkflowCleanup,
+	)
+}
+
+func (f *ExecutorFactoryBase) GetNewWorkflowDeleteManager(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+) deletemanager.DeleteManager {
+	return deletemanager.NewDeleteManager(
+		shardCtx,
+		workflowCache,
+		f.Config,
+		f.ArchivalClient,
+		shardCtx.GetTimeSource(),
+		f.VisibilityManager,
+	)
+}
+
+func (f *ExecutorFactoryBase) GetNewDefaultStandbyNDCHistoryResender(
```

Comment:
I think we should get rid of this and just make it a static function or move it to some other object. It's good that we're extracting common code, but I don't think it makes sense in `ExecutorFactory`

Reply:
I agree that it doesn't exactly belong here, but I think ideally we would refactor the queue factory logic as a whole after this change. I also considered just making this function a constant, but then we wouldn't be able to customize it in the future and I don't know if that is something we are likely to want or not.